plugin-server/src/worker/ingestion/timestamps.ts

```diff
@@ -1,37 +1,39 @@
 import { PluginEvent } from '@posthog/plugin-scaffold'
 import * as Sentry from '@sentry/node'
-import { StatsD } from 'hot-shots'
 import { DateTime, Duration } from 'luxon'
 
 import { status } from '../../utils/status'
 
-export function parseEventTimestamp(data: PluginEvent, statsd?: StatsD | undefined): DateTime {
+type InvalidTimestampCallback = (teamId: number) => void
+
+export function parseEventTimestamp(data: PluginEvent, callback?: InvalidTimestampCallback): DateTime {
     const now = DateTime.fromISO(data['now']).toUTC()
     const sentAt = data['sent_at'] ? DateTime.fromISO(data['sent_at']).toUTC() : null
 
     const parsedTs = handleTimestamp(data, now, sentAt)
     const ts = parsedTs.isValid ? parsedTs : DateTime.utc()
-    if (!parsedTs.isValid) {
-        statsd?.increment('process_event_invalid_timestamp', { teamId: String(data['team_id']) })
+    if (!parsedTs.isValid && callback) {
+        callback(data['team_id'])
     }
     return ts
 }
 
 function handleTimestamp(data: PluginEvent, now: DateTime, sentAt: DateTime | null): DateTime {
     if (data['timestamp']) {
-        if (sentAt) {
+        const timestamp = parseDate(data['timestamp'])
+        if (sentAt && sentAt.isValid && timestamp.isValid) {
             // sent_at - timestamp == now - x
             // x = now + (timestamp - sent_at)
             try {
                 // timestamp and sent_at must both be in the same format: either both with or both without timezones
                 // otherwise we can't get a diff to add to now
-                return now.plus(parseDate(data['timestamp']).diff(sentAt))
+                return now.plus(timestamp.diff(sentAt))
             } catch (error) {
                 status.error('⚠️', 'Error when handling timestamp:', error)
```

Comment:
while we're at it could you fix this one so we actually show the error

Reply:
Looks like a widespread issue, I'll fix it here by explicitly taking the `error.message` and will keep the wider investigation on the backburner for later
![image](https://user-images.githubusercontent.com/6241083/204814542-52567e3a-7c5c-4a2c-a73b-bbdb90b5aea8.png)